app/controllers/hostgroups_controller.rb

```diff
@@ -102,8 +103,31 @@ def process_hostgroup
     render :partial => "form"
   end
 
+  def taxonomy_warnings
+    @warnings ||= []
+
+    @warnings += organization_warnings(@hostgroup)
+    @warnings += location_warnings(@hostgroup)
+  end
+
   private
 
+  def organization_warnings(hostgroup)
+    warnings = []
+
+    organizations_count = @hostgroup.hosts.unscoped.distinct.count(:organization_id)
+    warnings << _('The hostgroup is used by hosts in multiple organizations') if organizations_count > 1
```

Comment:
I am thinking from security point of view: if the user is not authorized for an organization or host, the link will lead him to an empty page. That can be confusing.

That is without considering the fact that we will show the user ids of those unauthorized objects. I am not sure if we should be that paranoid though.